src/app/modules/recipe-list/containers/recipe-list.component.ts

```diff
@@ -39,17 +42,37 @@ export class RecipeListComponent implements OnInit {
     this.getImages();
     this.activeRoute.queryParams.subscribe(params => this.getAllRecipes({
       page: params.page || 1,
-      per_page: params.per_page || defaultPerPage
+      per_page: params.per_page || defaultPerPage,
+      sort: params.sort || defaultSortBy,
+      order: params.order || defaultOrder
     }))
   }
 
   public onPageChange(pageEvent: PageEvent): void {
     this.router.navigate(["./"], {
-      queryParams: {page: pageEvent.pageIndex + 1, per_page: pageEvent.pageSize},
+      queryParams: {page: pageEvent.pageIndex + 1, per_page: pageEvent.pageSize, sort: this.sort, order: this.order},
       queryParamsHandling: 'merge'
     })
   }
 
+  onSortChange(changedOption: string): void {
```

Comment:
```suggestion
  onSortChange(sort: string, order: string): void {
```
pass  sort and order to this function, or create two separate functions